lib/message.js

```diff
@@ -0,0 +1,44 @@
+'use babel';
+/* @flow */
+
+type Range = Array<Array<number>>
+type LinterTrace = { type: 'Trace', text?: string, html?: string, filePath?: string, range?: Range, class?: string }
+type LinterMessage = { type: string, text?: string, html?: string, filePath: string, range: Range, trace?: Array<LinterTrace> }
+type FlowMessage = { descr: string, level: string, path: string, start: number, end: number, line: number, endline: number }
+
+function extractRange(message: FlowMessage): Range {
+  return [
+    [message.line - 1, message.start - 1],
+    [message.endline - 1, message.end],
+  ];
+}
+
+function flowMessageToTrace(message: FlowMessage): LinterTrace {
+  return {
+    type: 'Trace',
+    text: message.descr,
+    filePath: message.path,
+    range: extractRange(message),
+  };
+}
+
+function flowMessageToLinterMessage({ message: messages }): LinterMessage {
+  const first = messages[0];
+
+  return {
+    type: first.level,
+    text: messages.map(msg => msg.descr).join('. '),
```

Comment:
It's because the first message description is often only one word and doesn't help understand the problem. concatenating message from traces create a understandable error message (to show in linter bottom pane)